weave/execute_fast.py

```diff
@@ -77,7 +77,9 @@ def _resolve_static_branches(map_fn):
                 return _resolve_static_branches(fixed)
 
         if all(isinstance(v, graph.ConstNode) for v in inputs.values()):
-            res = weave_internal.use(map_fn)
+            call_node = graph.OutputNode(map_fn.type, map_fn.from_op.name, inputs)
+            res = weave_internal.use(call_node)
```

Comment:
this is to bust the cache?

Reply:
No, before we were calling use on map_fn, which is the original node before we recursively evaluate its inputs. Basically totally wrong, we were recursively evaluating the root, then the root and its descendent, then root descendent, descendent-descendent and so on.